### ISSUE TITLE
What is a test?

### ISSUE THREAD

**Issue by gruninger** [gruninger/colore#11]
A test is a process whose effects are compared to the intended effects of a manufacturing process.

**Comment by plbt5** [comment]
In my experience a test is a (described) process whose effects are to
discover unintended behaviour in respect to (under)specified criteria.

Best
   Paul (former flight test instrumentation engineer)

Op vr 3 aug. 2018 17:12 schreef gruninger <notifications@github.com>:

**Comment by mhodki** [comment]
I would agree with Paul, Melinda

**Comment by ekwallace** [comment]
I would disagree.  My problem with Paul's definition above is that it is exclusively about _behavior_.  

I worked on a canonical data model a couple years ago to support exchange of messages and data elements for laboratory testing and product inspection.  The usual focus of these tests was to determine _characteristics_ of the material under test.  We described the results of such testing, for example a Certificate of Analysis, as: "Assertion of characteristics of a quantity of material by qualified laboratory or quality management personnel".   The process used for testing typically specialized standard test methods.  So using BFO-speak, an OAGIS:Test might be "A process following a specified plan that is used to determine specific bfo:qualities that inhere in the material under test, often against specific ranges of values of that quality."  

But some characteristics of interest may be qualities of dispositions.  For systems, dispositions include functions and capabilities (including behaviors), so we really do need to include both qualities of the material and qualities of its dispositions (as I understand it), to include all the things we potentially would want to test in a product or other material entity under test.

**Comment by ekwallace** [comment]
I should have mentioned.  OAGIS was the canonical information model standard that we extended to support exchange of test/inspection requests, test results, etc.

**Comment by schneidert** [comment]
Obviously there are different types of tests. I'd suggest looking at the common definitions of the word 'test' and determining if a) there is one that's appropriate (for IOF use), or b) if there's some additional characteristics or criteria that distinguish what the IOF intention of 'test' is (from the common definition).

Here are some common definitions:

a procedure intended to establish the quality, performance, or reliability of something, especially before it is taken into widespread use: no sparking was visible during the tests.

• a short written or spoken examination of a person's proficiency or knowledge: a spelling test.

• an event or situation that reveals the strength or quality of someone or something by putting them under strain: this is the first serious test of the peace agreement.

• an examination of part of the body or a body fluid for medical purposes, especially by means of a chemical or mechanical procedure rather than simple inspection: a test for HIV | eye tests.

• Chemistry a procedure employed to identify a substance or to reveal the presence or absence of a constituent within a substance.

• the result of a medical examination or analytical procedure: a positive test for protein.

• a means of establishing whether an action, item, or situation is an instance of a specified quality, especially one held to be undesirable: a statutory test of obscenity.
